backoff/exponential.go

```diff
@@ -377,7 +378,13 @@ func (e *exponentialBackoff) Enumerate(volumeIds []*string,
 	)
 	conditionFn := func() (bool, error) {
 		enumerateResponse, origErr = e.cloudOps.Enumerate(volumeIds, labels, setIdentifier)
-		msg := fmt.Sprintf("Failed to enumerate drives (%v).", volumeIds)
+		volumeIdsStr, err := volumeIdsStringDereference(volumeIds)
+		var msg string
+		if err != nil {
+			msg = err.Error()
```

Comment:
I don't think you should change the message here. Instead, you can print the err here but you should retain the message. In line 388 we are responding to the origErr and not err. So if origErr is not nil we need a message that says what are the drives that couldn't enumerate. If one of the drives happened to be nil, we should just print that and let the person debugging take a look at it.
Additionally, in line 401, instead of returning, you should continue and make that volume name to be "". I don't expect to have a nil volume id since we would most likely crash/catch that in line 380. Even if we somehow have a nil volume id, let's not stop printing the rest of them.

Reply:
good point. changed the check to continue when it sees a nil. and you are right, we should expect the cloudprovider to handle the nil case for us.